tests/ModelContextProtocol.Tests/Transport/InMemoryTransportTests.cs

```diff
@@ -0,0 +1,136 @@
+using Microsoft.Extensions.Logging.Abstractions;
+
+using ModelContextProtocol.Protocol.Messages;
+using ModelContextProtocol.Protocol.Transport;
+using ModelContextProtocol.Server;
+
+namespace ModelContextProtocol.Tests.Transport;
+
+public class InMemoryTransportTests
```

Comment:
Are there any existing tests it would make sense to migrate to the in-memory transport?

Also, it might be worth changing this to derive from LoggedTest and use the XunitLoggerProvider introduced by #81.